include/nbl/ui/CWindowWin32.h

```diff
@@ -39,97 +41,150 @@ class CWindowWin32 final : public IWindowWin32
+			wcex.hIconSm = 0;
+
+			RegisterClassEx(&wcex);
+
+			// calculate client size
+
+			RECT clientSize;
+			clientSize.top = 0;
+			clientSize.left = 0;
+			clientSize.right = _w;
+			clientSize.bottom = _h;
+
+			DWORD style = WS_POPUP; // TODO why popup?
+
+			if ((_flags & ECF_FULLSCREEN) == 0)
+			{
+				if ((_flags & ECF_BORDERLESS) == 0)
+				{
+					style |= WS_BORDER;
+					style |= (WS_SYSMENU | WS_CAPTION);
+				}
+				// ? not sure about those below
+				style |= WS_CLIPCHILDREN;
+				style |= WS_CLIPSIBLINGS;
+			}
+			if (_flags & ECF_MINIMIZED)
+			{
+				style |= WS_MINIMIZE;
+			}
+			if (_flags & ECF_MAXIMIZED)
 			{
-				style |= WS_BORDER;
-				style |= (WS_SYSMENU | WS_CAPTION);
+				style |= WS_MAXIMIZE;
+			}
+			if (_flags & ECF_ALWAYS_ON_TOP)
+			{
+				style |= WS_EX_TOPMOST;
+			}
+			if ((_flags & ECF_HIDDEN) == 0)
+			{
+				style |= WS_VISIBLE;
 			}
 
-			// ? not sure about those below
-			style |= WS_CLIPCHILDREN;
-			style |= WS_CLIPSIBLINGS;
-		}
-		if (isMinimized())
-			style |= WS_MINIMIZE;
-		if (isMaximized())
-			style |= WS_MAXIMIZE;
-		// TODO:
-		// if (hasMouseCaptured())
-		// if (hasInputFocus())
-		// if (hasMouseFocus())
-		// if (isAlwaysOnTop())
-
-		AdjustWindowRect(&clientSize, style, FALSE);
-
-		const int32_t realWidth = clientSize.right - clientSize.left;
-		const int32_t realHeight = clientSize.bottom - clientSize.top;
-
-		int32_t windowLeft = (GetSystemMetrics(SM_CXSCREEN) - realWidth) / 2;
-		int32_t windowTop = (GetSystemMetrics(SM_CYSCREEN) - realHeight) / 2;
-
-		if (windowLeft < 0)
-			windowLeft = 0;
-		if (windowTop < 0)
-			windowTop = 0;	// make sure window menus are in screen on creation
-
-		if (isFullscreen())
-		{
-			windowLeft = 0;
-			windowTop = 0;
-		}
 
-		// create window
+			// TODO:
+			// if (hasMouseCaptured())
+			// if (hasInputFocus())
+			// if (hasMouseFocus())
 
-		m_native = CreateWindow(classname, __TEXT(""), style, windowLeft, windowTop,
-			realWidth, realHeight, NULL, NULL, hinstance, NULL);
-		m_width = realWidth;
-		m_height = realHeight;
+			AdjustWindowRect(&clientSize, style, FALSE);
 
-		if (!isHidden())
-			ShowWindow(m_native, SW_SHOWNORMAL);
-		UpdateWindow(m_native);
+			const int32_t realWidth = clientSize.right - clientSize.left;
+			const int32_t realHeight = clientSize.bottom - clientSize.top;
 
-		// fix ugly ATI driver bugs. Thanks to ariaci
-		// TODO still needed?
-		MoveWindow(m_native, windowLeft, windowTop, realWidth, realHeight, TRUE);
-    }
+			int32_t windowLeft = (GetSystemMetrics(SM_CXSCREEN) - realWidth) / 2;
+			int32_t windowTop = (GetSystemMetrics(SM_CYSCREEN) - realHeight) / 2;
+
+			if (windowLeft < 0)
+				windowLeft = 0;
+			if (windowTop < 0)
+				windowTop = 0;	// make sure window menus are in screen on creation
+
+			if (_flags & ECF_FULLSCREEN)
+			{
+				windowLeft = 0;
+				windowTop = 0;
+			}
+
+			// create window
+
+			 nativeWindow = CreateWindow(classname, __TEXT(""), style, windowLeft, windowTop,
```

Comment:
TODO: add a creation parameter for the window title, and add a method to IWindow for changing the window title at runtime.

This also needs a capability query (because not every platform has window titles or even if they do they might not be changeable).